ui: fix db page Node/Regions column rendering

xinhaoz opened the pull request:
Previously, the db page was not updating its columns if the
`showNodeRegionsColumn` prop changed. The db details page was
also not filtering out the regions column when desired.

Epic: none

Release note (bug fix): node/regions columns in db and db details
page should properly render. This column is hidden for tenants and
otherwise is shown for clusters with > 1 node.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/105765)
<!-- Reviewable:end -->

xinhaoz committed:
ui: fix db page Node/Regions column rendering

Previously, the db page was not updating its columns if the
`showNodeRegionsColumn` prop changed. The db details page was
also not filtering out the regions column when desired.

Epic: none

Release note (bug fix): node/regions columns in db and db details
page should properly render. This column is hidden for tenants and
otherwise is shown for clusters with > 1 node.

THardy98 reviewed (changes requested):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>
left a small note/question

> Reviewed 2 of 2 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/105765#-:-NZ6P95r-xh9HCKV7ZXw:b-3w68ol)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @xinhaoz)

___
*[`pkg/ui/workspaces/cluster-ui/src/databasesPage/databasesPage.tsx` line 289 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/105765#-NZ6PC8I8MuG0UmKRLqj:-NZ6PC8I8MuG0UmKRLqk:b-ydd9y1) ([raw file](https://github.com/cockroachdb/cockroach/blob/039ac80ae80c64569f97feb2b46d4430e5a8eb43/pkg/ui/workspaces/cluster-ui/src/databasesPage/databasesPage.tsx#L289)):*
> ```XML
>       prevProps.indexRecommendationsEnabled !==
>         this.props.indexRecommendationsEnabled ||
>       prevProps.showNodeRegionsColumn !== this.props.showNodeRegionsColumn
> ```

In principle, the change here is saying we should render this column if the # of nodes or regions changes to >1. Was this the case that was causing the cypress test to fail?


<!-- Sent from Reviewable.io -->

THardy98 reviewed (approved):
oops meant to approve

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/105765#-:-NZ6Q1dJ4CHqLnZ0dcew:bl9kzbl)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @xinhaoz)



<!-- Sent from Reviewable.io -->

xinhaoz commented:
TFTR! Test failure unrelated to UI.
bors r+
